src/BLE_Client.cpp

```diff
@@ -212,12 +206,18 @@ bool SpinBLEClient::connectToServer()
             charUUID = FLYWHEEL_UART_TX_UUID;
             debugDirector("trying to connect to Flywheel Bike");
         }
-        else
+        else if (myDevice->isAdvertisingService(CYCLINGPOWERSERVICE_UUID))
```

Comment:
A later improvement is to pull this code up into the SensorData implementation so that contributing sources does not require changes all over the code